File: ccif/src/java/other/org/himalay/ccif/TimeZone.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:08 EST 2016")

public  class TimeZone extends BinStruct implements PublicBinMsg {




    // members variables
    // timeZoneOffset
    public int timeZoneOffset ;

    public TimeZone () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize timeZoneOffset

    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read timeZoneOffset
        {
            timeZoneOffset=istream.readUnsignedShort();
            retVal+=2;
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;



        // write timeZoneOffset
        ostream.writeShort(timeZoneOffset);
        retVal +=2;
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("TimeZone\n");
        dc.increaseIndent();
        int retVal= 0;
        // write timeZoneOffset
        dc.indent();
        dc.getPs().println("timeZoneOffset="+timeZoneOffset+"(0x"+ Integer.toHexString(timeZoneOffset) + ")" );
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for timeZoneOffset
    //public int getTimeZoneOffset()
    //{
    //    return timeZoneOffset ;
    //}


    // Setter for timeZoneOffset
    //public void setTimeZoneOffset(int val)
    //{
    //    this.timeZoneOffset= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/TimeZone.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "TimeZone.h"

#ifdef org_himalay_ccif_TimeZone__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
TimeZone::TimeZone() {
    references= 0;


    // timeZoneOffset
    timeZoneOffset= (ui16)0;
}
// Destructor
TimeZone::~TimeZone() {
    // timeZoneOffset

}

int TimeZone::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // timeZoneOffset
    {
        timeZoneOffset=istream.readUI16();
        retVal+=2;
    }
    return retVal;
}

int TimeZone::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read timeZoneOffset
    {
        timeZoneOffset=istream.readUI16();
        retVal+=2;
    }

    return retVal;
}

int TimeZone::write(DataOutputStream& ostream) { // throws IOException


    int retVal= 0;
    // write timeZoneOffset
    ostream.writeUI16(timeZoneOffset);
    retVal +=2;

    return retVal;
}

int TimeZone::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("TimeZone\n");
    dc.increaseIndent();
    int retVal= 0;
    // write timeZoneOffset
    dc.indent();
    dc.getPs().print("timeZoneOffset=");
    dc.getPs().println((long)timeZoneOffset);
    dc.decreaseIndent();
    return retVal;
}


// Getter for timeZoneOffset
//ui16 TimeZone::getTimeZoneOffset()
//{
//return this->timeZoneOffset;
//}

// Setter for timeZoneOffset
//void TimeZone::setTimeZoneOffset(ui16 val)
//{
//this->timeZoneOffset= val;
//}

int TimeZone::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_TimeZone__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::TimeZone* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::TimeZone* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code